Habitica/src/main/java/com/habitrpg/android/habitica/helpers/PopupNotificationsManager.java

```diff
@@ -0,0 +1,134 @@
+
+        SimpleDraweeView imageView = (SimpleDraweeView) view.findViewById(R.id.imageView);
+        String imageKey = "inventory_present_11";
+        if (notification.data.rewardKey != null) {
+            imageKey = notification.data.rewardKey.get(0);
+
+            if (notification.data.reward != null && notification.data.reward.size() > 0) {
+                String earnedString = "";
+                int count = 0;
+                for (Reward reward : notification.data.reward) {
+                    earnedString += reward.key;
+                    count += 1;
+                    if (notification.data.reward.size() > 1 && count != notification.data.reward.size()) {
+                        earnedString += ", ";
+                    }
+                }
+                youEarnedMessage = "You earned a " + earnedString + " as a reward for your devotion to improving your life.";
+            }
+        }
+        DataBindingUtils.loadImage(imageView, imageKey);
+
+        String message = "Your next prize unlocks at " + notification.data.nextRewardAt + " Check-Ins";
+        TextView nextUnlockTextView = (TextView) view.findViewById(R.id.next_unlock_message);
+        nextUnlockTextView.setText(message);
+
+        TextView youEarnedTexView = (TextView) view.findViewById(R.id.you_earned_message);
+        youEarnedTexView.setText(youEarnedMessage);
+
+        Button confirmButton = (Button) view.findViewById(R.id.confirm_button);
+
+        AlertDialog.Builder builder = new AlertDialog.Builder(HabiticaApplication.currentActivity)
+                .setTitle(title)
+                .setView(view)
+                .setMessage("");
+
+        final AlertDialog dialog = builder.create();
+        dialog.show();
+
+        confirmButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                if (apiHelper != null) {
+                    // @TODO: This should be handled somewhere else? MAybe we notifiy via event
+                    apiHelper.apiService.readNotificaiton(notification.getId())
+                            .compose(apiHelper.configureApiCallObserver())
+                            .subscribe(next -> {
+                            }, throwable -> {
+                            });
+                }
+
+                dialog.hide();
+            }
+        });
+
+        return true;
+    }
+
+    public Boolean showNotificationDialog(final List<Notification> notifications) {
+        if (notifications.size() == 0) {
+            return false;
+        }
+
+        HabiticaApplication.currentActivity.runOnUiThread(() -> {
+            if ((HabiticaApplication.currentActivity).isFinishing()) return;
+
+            if (this.seenNotifications == null) {
+                this.seenNotifications = new HashMap<>();
+            }
+
+            for (Notification notification: notifications) {
```

Comment:
Sure, but that shouldn't be an issue here since a user can only have one login incentive notification at a time and this only create dialogues for login notifications.